### PR TITLE
add manually input overlap matrix feature with example

### LINKED ISSUE #254: Add overlap matrix input for band plot function.
### Background

For models containing overlap matrices, sometimes the positive definite requirement of the overlapping matrix makes the accuracy requirement too high and it is difficult to reach the positive definite criterion. Therefore it is desired to provide a method of inputting the overlapping matrix from the outside to avoid overtraining of the model for this reason.

### Describe the solution you'd like

Read in overlaps.h5, and override the overlap blocks in band construction.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Lonya0**
This pull request introduces functionality to override the overlap matrix used in band structure calculations, along with associated updates to the codebase and test data. The most significant changes include adding support for the `override_overlap` parameter in multiple methods, refactoring code to integrate this functionality, and updating test data files to validate the new feature.

### Core functionality updates:

* **Support for overriding overlap matrix**:
  - Added the `override_overlap` parameter to `get_data` and `get_eigs` methods in `dptb/postprocess/elec_struc_cal.py`. This parameter allows specifying a custom overlap matrix file (`overlap.h5`) and ensures proper handling of edge and node overlap keys during calculations. [[1]](diffhunk://#diff-93b42b7711d165125a6b813a35c4522ee6b432baca1eab211dd0c91dc847df08L69-R76) [[2]](diffhunk://#diff-93b42b7711d165125a6b813a35c4522ee6b432baca1eab211dd0c91dc847df08R142-R164) [[3]](diffhunk://#diff-93b42b7711d165125a6b813a35c4522ee6b432baca1eab211dd0c91dc847df08L142-R178)
  - Integrated logic to validate the presence of the overlap file, extract overlap matrices, and apply them using the `block_to_feature` function. Warnings are logged if conflicts arise between the model's overlap matrix and the overridden one.

* **Refactoring and imports**:
  - Added `os` and `h5py` imports to handle file operations for the overlap matrix. Also imported `block_to_feature` from `dptb.data` for processing overlap blocks. [[1]](diffhunk://#diff-93b42b7711d165125a6b813a35c4522ee6b432baca1eab211dd0c91dc847df08R1-R2) [[2]](diffhunk://#diff-93b42b7711d165125a6b813a35c4522ee6b432baca1eab211dd0c91dc847df08L10-R12)

### Test data updates:

* **New data files for testing**:
  - Added comprehensive test data files under `dptb/tests/data/e3_band/data/Si64.0/` to validate the `override_overlap` functionality. These include atomic numbers (`atomic_numbers.dat`), basis (`basis.dat`), cell parameters (`cell.dat`), periodic boundary conditions (`pbc.dat`), atomic positions (`positions.dat`), and a VASP file (`Si64.vasp`). [[1]](diffhunk://#diff-012f419a4b3680e37913f22689b11c78e8bde2cebcf7dcae7b7786851da2b7e5R1-R64) [[2]](diffhunk://#diff-1b9e1ce5fbf2e184ce33faca714712e92f3de18fe202eb01436403b949b82d9fR1) [[3]](diffhunk://#diff-36a0b3b060c0a3c294849ef8333c1b6e1d6f64689df3c6842db0c8c66fca437aR1-R3) [[4]](diffhunk://#diff-5ac3e9c5c2b56b32c4bd44ce938174285268ac816dd5ea8cff8e787b15f1e18cR1-R3) [[5]](diffhunk://#diff-f9a901c087c7c0aaa91eee5d829877bee33694953a3648a2b0b0c27904b9de6aR1-R64) [[6]](diffhunk://#diff-4627789df1d9e6b43771940436055ac9f9ff151ae08d9bce5ce04320959f093cR1-R72)
  - Added `info.json` to provide metadata for the test data, specifying frame count, position type, and periodic boundary conditions.
 
Fix #254

### CODE REVIEW

File: dptb/postprocess/elec_struc_cal.py [hunk 1]
```diff
@@ -136,10 +145,36 @@ def get_data(self,data: Union[AtomicData, ase.Atoms, str],pbc:Union[bool,list]=N
         data = AtomicData.to_AtomicDataDict(data.to(device))
         data = self.model.idp(data)
```
Comment (AsymmetryChou) [inline dptb/postprocess/elec_struc_cal.py:147]: 建议加一个log.warning: 使用输入的overlap会覆盖model自带的overlap

File: examples/e3/band_plot_from_example.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+model = build_model(checkpoint="./ref_model/nnenv.ep1474.pth")
+
+ # build the dataset from the model
+dataset = build_dataset.from_model(
+    model=model,
+    root="./data",
+    prefix="Si64"
+)  
+
+jdata = j_loader("../../../../Desktop/DeePTB-main/DeePTB-main/examples/e3/band.json")
```
Comment (AsymmetryChou) [inline examples/e3/band_plot_from_example.py:16]: 这个路径需要改一下

File: dptb/postprocess/elec_struc_cal.py [hunk 3]
```diff
@@ -136,10 +145,36 @@ def get_data(self,data: Union[AtomicData, ase.Atoms, str],pbc:Union[bool,list]=N
...
 
+        if isinstance(override_overlap, str):
+            assert os.path.exists(override_overlap), "Overlap file not found."
+            overlap_blocks = h5py.File(override_overlap, "r")
+            if len(overlap_blocks) != 1:
+                log.info('Overlap file contains more than one overlap matrix, only first will be used.')
+            data[AtomicDataDict.NODE_OVERLAP_KEY] = None
+            data[AtomicDataDict.EDGE_OVERLAP_KEY] = None
+            if "0" in overlap_blocks:
```
[the remark this answers is not in the document]
Comment (AsymmetryChou) [reply dptb/postprocess/elec_struc_cal.py:155]: 你确认就行

[the remark this answers is not in the document]
Comment (AsymmetryChou) [reply dptb/postprocess/elec_struc_cal.py:155]: 另外，copilot提的建议我粗略看了一下，有些还挺有道理，斟酌修改一下吧~

**Comment by QG-phy** [comment]
> 更新了单元测试，以及功能的修复

单元测试没通过。

**Review by QG-phy — APPROVED** [review]
